place_tracker/lib/place_map.dart

```diff
@@ -0,0 +1,441 @@
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+
+import 'place.dart';
+import 'place_stub_data.dart';
+import 'place_details.dart';
+
+class PlaceMap extends StatefulWidget {
+  const PlaceMap({
+    Key key,
+    this.center,
+  }) : super(key: key);
+
+  final LatLng center;
+
+  @override
+  PlaceMapState createState() => PlaceMapState();
+}
+
+class PlaceMapState extends State<PlaceMap> {
+
+  static BitmapDescriptor _getPlaceMarkerIcon(PlaceCategory category) {
+    // TODO(kenzieschmoll): use custom marker assets.
+    double markerHue;
+    switch (category) {
+      case PlaceCategory.favorite:
+        markerHue = BitmapDescriptor.hueRed;
+        break;
+      case PlaceCategory.visited:
+        markerHue = BitmapDescriptor.hueViolet;
+        break;
+      case PlaceCategory.wantToGo:
+      default:
+        markerHue = BitmapDescriptor.hueAzure;
+    }
+    return BitmapDescriptor.defaultMarkerWithHue(markerHue);
+  }
+
+  static List<Place> _getPlacesForCategory(
+      PlaceCategory category, Map<Marker, Place> places) {
+    return places.values.where(
+            (Place place) => place.category == category).toList();
+  }
+
+  GoogleMapController mapController;
+  PlaceCategory _selectedPlaceCategory = PlaceCategory.favorite;
+  Map<Marker, Place> _places = Map<Marker, Place>();
+  Marker _pendingMarker;
+
+  void onMapCreated(GoogleMapController controller) async {
+    mapController = controller;
+    mapController.onInfoWindowTapped.add(_onInfoWindowTapped);
+
+    // Add stub data on creation so we have something interesting to look at.
+    final Map<Marker, Place> places = await _initializeStubPlaces();
+    _zoomToFitPlaces(
+      _getPlacesForCategory(_selectedPlaceCategory, places),
+    );
+  }
+
+  Future<Map<Marker, Place>> _initializeStubPlaces() async {
+    for (Place place in PlaceStubData.places) {
```

Comment:
It's a little more efficient to use `Future.wait` to wait for a list of Futures to complete.
```
await Future.wait PlaceStubData.places.map((Place place) => _initializeStubPlace(place)).toList();
```

Reply:
Done.